Build warnings in latest master

```
In file included from /usr/include/c++/14/cassert:44,
                 from /home/dale/git/overte/overte-master/libraries/shared/src/MovingMinMaxAvg.h:15,
                 from /home/dale/git/overte/overte-master/libraries/audio/src/InboundAudioStream.h:25,
                 from /home/dale/git/overte/overte-master/libraries/audio/src/MixedProcessedAudioStream.h:15,
                 from /home/dale/git/overte/overte-master/libraries/audio-client/src/AudioClient.h:40,
                 from /home/dale/git/overte/overte-master/interface/src/avatar/MyAvatar.cpp:34:
/home/dale/git/overte/overte-master/interface/src/avatar/MyAvatar.cpp: In member function ‘void MyAvatar::FollowHelper::deactivate(CharacterController::FollowType)’:
/home/dale/git/overte/overte-master/interface/src/avatar/MyAvatar.cpp:5771:22: warning: comparison is always true due to limited range of data type [-Wtype-limits]
 5771 |     assert((int)type >= 0 && (int)type < static_cast<int>(CharacterController::FollowType::Count));
      |            ~~~~~~~~~~^~~~
/home/dale/git/overte/overte-master/interface/src/avatar/MyAvatar.cpp: In member function ‘void MyAvatar::FollowHelper::activate(CharacterController::FollowType, bool)’:
/home/dale/git/overte/overte-master/interface/src/avatar/MyAvatar.cpp:5779:22: warning: comparison is always true due to limited range of data type [-Wtype-limits]
 5779 |     assert((int)type >= 0 && (int)type < static_cast<int>(CharacterController::FollowType::Count));
      |            ~~~~~~~~~~^~~~
/home/dale/git/overte/overte-master/interface/src/avatar/MyAvatar.cpp: In member function ‘bool MyAvatar::FollowHelper::isActive(CharacterController::FollowType) const’:
/home/dale/git/overte/overte-master/interface/src/avatar/MyAvatar.cpp:5786:22: warning: comparison is always true due to limited range of data type [-Wtype-limits]
 5786 |     assert((int)type >= 0 && (int)type < static_cast<int>(CharacterController::FollowType::Count));
      |            ~~~~~~~~~~^~~~
```

--- a/interface/src/avatar/MyAvatar.cpp
+++ b/interface/src/avatar/MyAvatar.cpp
@@ -5768,7 +5768,7 @@ void MyAvatar::FollowHelper::deactivate() {
 }
 
 void MyAvatar::FollowHelper::deactivate(CharacterController::FollowType type) {
-    assert((int)type >= 0 && (int)type < static_cast<int>(CharacterController::FollowType::Count));
+    assert(type < CharacterController::FollowType::Count);
     _timeRemaining[(int)type] = 0.0f;
 }
 
@@ -5776,14 +5776,14 @@ void MyAvatar::FollowHelper::deactivate(CharacterController::FollowType type) {
 // eg. activate(FollowType::Rotation, true) snaps the FollowHelper's rotation immediately
 // to the rotation of its _followDesiredBodyTransform.
 void MyAvatar::FollowHelper::activate(CharacterController::FollowType type, const bool snapFollow) {
-    assert((int)type >= 0 && (int)type < static_cast<int>(CharacterController::FollowType::Count));
+    assert(type < CharacterController::FollowType::Count);
 
     // TODO: Perhaps, the follow time should be proportional to the displacement.
     _timeRemaining[(int)type] = snapFollow ? CharacterController::FOLLOW_TIME_IMMEDIATE_SNAP : FOLLOW_TIME;
 }
 
 bool MyAvatar::FollowHelper::isActive(CharacterController::FollowType type) const {
-    assert((int)type >= 0 && (int)type < static_cast<int>(CharacterController::FollowType::Count));
+    assert(type < CharacterController::FollowType::Count);
     return _timeRemaining[(int)type] > 0.0f;
 }
 

--- a/libraries/physics/src/CharacterController.h
+++ b/libraries/physics/src/CharacterController.h
@@ -54,7 +54,7 @@ class CharacterController : public btCharacterControllerInterface {
 
 public:
     enum class FollowType : uint8_t {
-        Rotation,
+        Rotation = 0,
         Horizontal,
         Vertical,
         Count

